src/boot/bootctl-install.c

```diff
@@ -423,12 +460,192 @@ static int copy_one_file(const char *esp_path, const char *name, bool force) {
         return ret;
 }
 
-static int install_binaries(const char *esp_path, const char *arch, bool force) {
+/* Parse shim's boot.csv https://github.com/rhboot/shim/blob/main/README.fallback
+ * Only parse file name and description in the first line. That's the only relevant information for
+ * systemd-boot. */
+static int parse_boot_csv_mem(
+                const char16_t *csv,
+                size_t size /* bytes */,
+                char **ret_loader_path,
+                char **ret_description) {
+
+        _cleanup_free_ char *buf = NULL, *loader_path = NULL, *description = NULL;
+        const char *s;
+        int r;
+
+        assert(csv);
+
+        /* Enforce unicode LE to avoid decoding garbage */
+        if (*csv != 0xfeff || size < 2)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Invalid boot csv: %hx", *csv);
+
+        buf = utf16_to_utf8(csv+1, size-2);
+        if (!buf)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to decode boot csv. Must be UTF-16");
+
+        /* Only care about first line */
+        s = truncate_nl(buf);
+
+        r = extract_many_words(&s, ",", EXTRACT_KEEP_QUOTE, &loader_path, &description);
+        if (r < 0)
+                return log_error_errno(r, "Failed to parse boot csv: %m");
+        if (r < 2)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Missing file name and description in boot csv");
```

Comment:
I don't particularly care either way, but please use dots or don't.

Instead of saying "boot csv" it'd be much more user friendly to give the path name…